builder/virtualbox/ovf/config_test.go

```diff
@@ -110,3 +112,47 @@ func TestNewConfig_shutdown_timeout(t *testing.T) {
 		t.Fatalf("bad: %s", err)
 	}
 }
+
+// TestChecksumFileNameMixedCaseBug reproduces Github issue #9049:
+//	https://github.com/hashicorp/packer/issues/9049
+func TestChecksumFileNameMixedCaseBug(t *testing.T) {
```

Comment:
I think for this particular issue you just want to verify that the value for `config["checksum"]` is unmodified after calling the `c.Prepare` as the process that calls the actual download step is in the builder's `Run` method (i.e `common.StepDownload`). 

I would suggest writing a table test here with two test cases one with the checksum path is all lowercase and one which is Mixed case and validate that those values are not change. 

Something like
```go

func TestChecksumFileNameMixedCaseBug(t *testing.T) {                                                                                              
  tt := []struct {                                                                                                                                 
    Name         string                                                                                                                            
    ChecksumPath string                                                                                                                            
  }{                                                                                                                                               
    {"LowerCase", "/tmp/random/file.md5"},                                                                                                         
    {"MixedCase", "/tmp/random/FILE.md5"},                                                                                                         
  }                                                                                                                                                
                                                                                                                                                   
  for _, tc := range tt {                                                                                                                          
    tc := tc                                                                                                                                       
                                                                                                                                                   
    cfg := testConfig(t)                                                                                                                           
    cfg["source_path"] = "somerandom.ovf"                                                                                                                 
    cfg["checksum_type"] = "file"                                                                                                                  
    cfg["checksum"] = tc.ChecksumPath                                                                                                              
    cfg["type"] = "virtualbox-ovf"                                                                                                                 
                                                                                                                                                   
    var c Config                                                                                                                                   
    warns, err := c.Prepare(cfg)                                                                                                                   
    if err != nil {                                                                                                                                
      t.Errorf("config failed to Prepare, %s", err.Error())                                                                                        
      t.Fail()                                                                                                                                     
    }                                                                                                                                              
                                                                                                                                                   
    if len(warns) != 0 {                                                                                                                           
      t.Errorf("Encountered warnings during config preparation: %s", warns)                                                                        
    }                                                                                                                                              
                                                                                                                                                   
    if c.Checksum != tc.ChecksumPath {                                                                                                                   t.Errorf("expected Checksum path to be %q, but got %q after calling Prepare", tc.ChecksumPath, c.Checksum)                                       }                                                                                                                                              
  }                                                                                                                                                
                                                                                                                                                   
} 
```


This test will fail on master which is what we want to reproduce the bug and should pass with this change in place. 

As for testing the download capability you can add an acceptance test that runs the builders against virtualbox. You can copy this [file](https://github.com/hashicorp/packer/blob/master/builder/virtualbox/iso/builder_acc_test.go) into the ovf directory alongside the builder.go file and update it to use the ovf builder and give it a source path and checksum file that is publicly available so that others who want to run the test can run it. That might be the tricky part and probably why we don't currently have an acceptance test for it.


In cases like this we can ask the reporter on the issue to test  and validate that the change fixes the issue. There are test binaries created from this PR available at https://app.circleci.com/pipelines/github/hashicorp/packer/4720/workflows/2fbf30a5-bb92-4de6-8820-c6b2d7ba36db/jobs/47204/steps

Reply:
A table test is a good idea, thanks. 

I'm not sure the value in the acceptance test is really there then, would you disagree?